tools/stonith_admin.c

```diff
@@ -428,18 +387,11 @@ validate(stonith_t *st, const char *agent, const char *id,
     rc = st->cmds->validate(st, st_opt_sync_call, id, NULL, agent, params,
                             timeout, &output, &error_output);
 
-    if (!quiet) {
-        printf("Validation of %s %s\n", agent, (rc? "failed" : "succeeded"));
-        if (output && *output) {
-            puts(output);
-            free(output);
-        }
-        if (error_output && *error_output) {
-            puts(error_output);
-            free(error_output);
-        }
+    if (quiet) {
+        return rc;
     }
-    return rc;
+
+    return out->message(out, "validate", agent, id, output, error_output, rc); 
```

Comment:
We should still return rc; the caller cares about the result of the validation itself